lib/packwerk/cli/uses_parse_run.rb

```diff
@@ -0,0 +1,91 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "optparse"
+
+module Packwerk
+  class Cli
+    module UsesParseRun
+      extend T::Sig
+      extend T::Helpers
+
+      requires_ancestor { Cli::BaseCommand }
+
+      sig do
+        params(
+          args: T::Array[String],
+          configuration: Configuration,
+          out: T.any(StringIO, IO),
+          err_out: T.any(StringIO, IO),
+          progress_formatter: Formatters::ProgressFormatter,
+          offenses_formatter: OffensesFormatter,
+        ).void
+      end
+      def initialize(args, configuration:, out:, err_out:, progress_formatter:, offenses_formatter:)
+        @args = args
+        @configuration = configuration
+        super
+        @parsed_options = T.let(parse_options, T::Hash[Symbol, T.untyped])
+        @configuration.parallel = @parsed_options[:parallel]
+        @files_for_processing = T.let(fetch_files_to_process, FilesForProcessing)
+        @offenses_formatter = T.let(find_offenses_formatter || @offenses_formatter, OffensesFormatter)
+      end
+
+      private
+
+      sig { returns(FilesForProcessing) }
+      def fetch_files_to_process
+        FilesForProcessing.fetch(
+          relative_file_paths: @parsed_options[:relative_file_paths],
+          ignore_nested_packages: @parsed_options[:ignore_nested_packages],
+          configuration: @configuration
+        )
+      end
+
+      sig { returns(T.nilable(OffensesFormatter)) }
+      def find_offenses_formatter
+        if @parsed_options[:formatter_name]
+          OffensesFormatter.find(@parsed_options[:formatter_name])
+        end
+      end
+
+      sig { returns(ParseRun) }
+      def parse_run
+        ParseRun.new(
+          relative_file_set: @files_for_processing.files,
+          parallel: @configuration.parallel?,
+        )
+      end
+
+      sig { returns(T::Hash[Symbol, T.untyped]) }
+      def parse_options
```

Comment:
Thinking ahead slightly, it might make sense for the base class to implement `def options; raise NotImplementedError; end` and have this be a memoized public method for commands to implement.